pkg/model/manifest_friendly_err.go

```diff
@@ -0,0 +1,165 @@
+	manifestKeys["model.DevRC"] = manifestKeys["model.Dev"]
+	manifestKeys["model.devType"] = manifestKeys["model.Dev"]
+
+	for structName, structKeywords := range manifestKeys {
+		for _, keyword := range structKeywords {
+			// example: line 5: field contex not found in type model.buildInfoRaw
+			// (.*?): this excludes eerything before the keyword "field"
+			// (\w+): this captures the keyword we want to calculate the levenshtein distance with
+			// (in type|into): this ensures to match all variations of the error message
+			// (.*?): this excludes everything after the message that we want to find
+			pattern := fmt.Sprintf(`(.*?)field (\w+) not found (in type|into) %s(.*?)`, structName)
+
+			// keywordInGroup is the index of the capturing group that contains the actual mistyped keyword
+			// set to 2 because index 0 is the whole sentence and index 1 is "line 5"
+			keywordInGroup := 2
+			rule := suggest.NewLevenshteinRule(pattern, keyword, keywordInGroup)
+			rules = append(rules, rule)
+		}
+	}
+
+	rules = append(rules,
+		// invalid properties
+		fieldsNotExistingRule(),
+
+		// struct names
+		suggest.NewStrReplaceRule("in type model.manifestRaw", "the okteto manifest"),
+		suggest.NewStrReplaceRule("in type model.ManifestBuild", "the 'build' section"),
+		suggest.NewStrReplaceRule("into model.ManifestBuild", "into a 'build' object"),
+		suggest.NewStrReplaceRule("in type model.buildInfoRaw", "the 'build' object"),
+		suggest.NewStrReplaceRule("in type model.devType", "the 'dev' object"),
+		suggest.NewStrReplaceRule("into model.devType", "the 'dev' object"),
+
+		// yaml types
+		suggest.NewStrReplaceRule(yaml.NodeTagSeq, "list"),
+		suggest.NewStrReplaceRule(yaml.NodeTagString, "string"),
+		suggest.NewStrReplaceRule(yaml.NodeTagBool, "boolean"),
+		suggest.NewStrReplaceRule(yaml.NodeTagInt, "integer"),
+		suggest.NewStrReplaceRule(yaml.NodeTagFloat, "float"),
+		suggest.NewStrReplaceRule(yaml.NodeTagMap, "object"),
+
+		// misc
+		suggest.NewStrReplaceRule("yaml: unmarshal errors:\n", ""),
+		suggest.NewStrReplaceRule("invalid manifest: ", ""),
+		indentNumLines(),
+	)
+
+	return rules
+}
+
+func isYamlErrorWithoutLinkToDocs(err error) bool {
+	return isYamlError(err) && !hasLinkToDocs(err)
+}
+
+// hasLinkToDocs checks if the error already contains a link to our docs
+func hasLinkToDocs(err error) bool {
+	return strings.Contains(err.Error(), "https://www.okteto.com/docs")
+}
+
+// isYamlError check if the error is related to YAML unmarshalling
+func isYamlError(err error) bool {
+	if err == nil {
+		return false
+	}
+	// detect yaml errors by prefix
+	if strings.Contains(err.Error(), "yaml:") {
+		return true
+	}
+	// detect yaml errors by type
+	var typeError *yaml.TypeError
+	return errors.As(err, &typeError)
+}
+
+func addYamlParseErrorHeading() *suggest.Rule {
+	addUrl := func(e error) error {
+		errorWithUrlToDocs := fmt.Sprintf("Your okteto manifest is not valid, please check the following errors:\n%s", e.Error())
```

Comment:
should we wrap all manifest errors with `ErrInvalidManifest` ? If not, some checks for this error will never work. For example -> https://github.com/okteto/okteto/blob/master/cmd/build/command.go#L138

Reply:
Good call @AdrianPedriza 👏 I've made the change here: https://github.com/okteto/okteto/pull/4009/commits/ffe965d4ae9d44b3bffa9161ee1ecf0ccb8f75a4